service/Abstractions/KernelMemoryBuilderExtensions.cs

```diff
@@ -137,6 +138,17 @@ public static partial class KernelMemoryBuilderExtensions
         return builder;
     }
 
+    public static IKernelMemoryBuilder WithCustomEmbeddingGenerator<T>(
+           this IKernelMemoryBuilder builder) where T : class, ITextEmbeddingGenerator
+    {
+        builder.Services.AddSingleton<ITextEmbeddingGenerator>(serviceProvider =>
+        {
+            return ActivatorUtilities.CreateInstance<T>(serviceProvider);
+        });
+
```

Comment:
isn't this equivalent to `builder.Services.AddSingleton<ITextEmbeddingGenerator, T>()` ?

Reply:
You're right, I used your original suggestion from https://github.com/microsoft/kernel-memory/issues/334, but actually we don't need the `ActivatorUtilities`.